lib/web_ui/lib/src/engine/canvaskit/surface.dart

```diff
@@ -160,12 +160,26 @@ class Surface {
       ..height = '${logicalHeight}px';
   }
 
+  void _contextLostListener(html.Event event) {
+    print('Flutter: restoring WebGL context.');
+    _forceNewContext = true;
+    // Force the framework to rerender the frame.
+    EnginePlatformDispatcher.instance.invokeOnMetricsChanged();
+    event.stopPropagation();
+    event.preventDefault();
+  }
+
   /// This function is expensive.
   ///
   /// It's better to reuse surface if possible.
   CkSurface _createNewSurface(ui.Size physicalSize) {
     // Clear the container, if it's not empty. We're going to create a new <canvas>.
     this.htmlCanvas?.remove();
+    //this.htmlCanvas?.removeEventListener(
+    //      'webglcontextlost',
+    //      _contextLostListener,
```

Comment:
Do we use `_contextLostListener` anywhere?

Reply:
Updated this part.